conda/cli/conda_argparse.py

```diff
@@ -85,17 +88,52 @@ def generate_parser():
     configure_parser_update(sub_parsers, name='upgrade')
     configure_parser_notices(sub_parsers)
 
+    # Add plugin subcommands
+    for plugin_sub in plugin_subcommands:
+        plugin_sub.add_argument_parser(sub_parsers)
+
     return p
 
 
 def do_call(args, parser):
-    relative_mod, func_name = args.func.rsplit('.', 1)
-    # func_name should always be 'execute'
-    from importlib import import_module
-    module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
+    """
+    Attempts several methods to call the appropriate sub-command:
+
+    If the ``func`` attribute on the parser object is a set to a string,
+    we assume a relatie namespace import.
+
+    If the ``func`` attribute is actually callable, we call it passing in
+    the ``args`` and the ``parser`` passed into this function.
+
+    Finally, we raise CondaError when none of this conditions are met.
+
+    """
+    if not hasattr(args, "func"):
+        print(
+            "This is a helpful error message for plugin authors! This happens when you do not "
+            "appropriately set the p.set_defaults(func=entry_point_function)"
+        )
+
+        sys.exit(1)
+
+    if isinstance(args.func, str):
+        relative_mod, func_name = args.func.rsplit(".", 1)
+        # func_name should always be 'execute'
+        from importlib import import_module
 
-    return getattr(module, func_name)(args, parser)
+        module = import_module(relative_mod, __name__.rsplit(".", 1)[0])
 
+        return getattr(module, func_name)(args, parser)
+
+    elif callable(args.func):
+        args.func(args, parser)
+
+    else:
+        raise CondaError(
+            "There is something mis-configured with your subcommands. This could be due to an "
+            "incorrectly configured plugin. Please try uninstalling any problematic plugins "
+            "you may have to remove this error."
```

Comment:
Would it help to print the `args` here so users can identify what is not working?

Reply:
Yeah, definitely would be nice to have some detailed debugging information here just like the error report we normally show.